core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/JWTGenerator.java

```diff
@@ -34,19 +37,27 @@
  */
 public final class JWTGenerator {
 
+	@Nullable
 	private final Algorithm algorithm;
+	@Nullable
+	private final RSAKeyProvider keyProvider;
+	
 	private final String issuer;
 	private final String emailClaimProperty;
 	private final String permissionsClaimProperty;
 
-	public JWTGenerator(final Algorithm algorithm, final String issuer, final String emailClaimProperty, final String permissionsClaimProperty) {
+	public JWTGenerator(final Algorithm algorithm, final RSAKeyProvider keyProvider, final String issuer, final String emailClaimProperty, final String permissionsClaimProperty) {
 		this.algorithm = algorithm;
+		this.keyProvider = keyProvider;
 		this.issuer = issuer;
 		this.emailClaimProperty = emailClaimProperty;
 		this.permissionsClaimProperty = permissionsClaimProperty;
 	}
 	
 	public String generate(String email, Map<String, Object> claims) {
+		if (algorithm == null || keyProvider != null && keyProvider.getPrivateKey() == null) {
```

Comment:
I'd split this check in two, with a different error message for each case:
- `algorithm == null` &rarr; `"No algorithm specified, cannot sign"`
- `keyProvider != null && keyProvider.getPrivateKey() == null` &rarr; `"Algorithm requires private key but none given, cannot sign"`

Reply:
Hmm, both represent the same thing for me, token signing is not configured, aka disabled, aka user cannot generate a token.

Reply:
Another alternative would be to introduce an interface for JWTGenerator and use a different implementation when the service is disabled.

Reply:
Just to help out with deciphering the error message 😄

Reply:
Refactored the relevant code statements and classes to be more expressive and easier to read and understand. 🎶